tiatoolbox/models/segmentation/semantic_segmentor.py

```diff
@@ -0,0 +1,983 @@
+            self.reader = self._get_reader(self.wsi_paths[self.wsi_idx])
+
+        # this is in XY and at requested resolution not baseline
+        bound = self.mp_shared_space.patch_inputs[idx]
+        bound = bound.numpy()  # expected to be torch.Tensor
+
+        # be the same as bounds br-tl, unless bounds are of float
+        patch_data_ = []
+        scale_factors = self.ioconfig.scale_to_highest(
+            self.ioconfig.input_resolutions, self.ioconfig.resolution_unit
+        )
+        for idy, resolution in enumerate(self.ioconfig.input_resolutions):
+            resolution_bound = np.round(bound * scale_factors[idy])
+            patch_data = self.reader.read_bounds(
+                resolution_bound.astype(np.int32),
+                coord_space="resolution",
+                pad_constant_values=0,  # expose this ?
+                **resolution,
+            )
+
+            if self.preproc is not None:
+                patch_data = patch_data.copy()
+                patch_data = self.preproc(patch_data)
+            patch_data_.append(patch_data)
+        if len(patch_data_) == 1:
+            patch_data_ = patch_data_[0]
+
+        bound = self.mp_shared_space.patch_outputs[idx]
+        return patch_data_, bound
+
+
+class SemanticSegmentor:
+    """Pixel-wise segmentation predictor.
+
+    Note, if model is supplied in the arguments, it will override the backbone.
+
+    Args:
+        model (nn.Module): Use externally defined PyTorch model for prediction with.
+            weights already loaded. Default is `None`. If provided,
+            `pretrained_model` argument is ignored.
+        pretrained_model (str): Name of the existing models support by tiatoolbox
+            for processing the data. Refer to
+            `tiatoolbox.models.classification.get_pretrained_model` for details.
+            By default, the corresponding pretrained weights will also be
+            downloaded. However, you can override with your own set of weights
+            via the `pretrained_weight` argument. Argument is case insensitive.
+        pretrained_weight (str): Path to the weight of the corresponding
+            `pretrained_model`.
+        batch_size (int) : Number of images fed into the model each time.
+        num_loader_worker (int) : Number of workers to load the data.
+            Take note that they will also perform preprocessing.
+        num_postproc_worker (int) : This value is there to maintain input
+            compatibility with `tiatoolbox.models.classification` and is
+            not used.
+        verbose (bool): Whether to output logging information.
+        dataset_class (obj): Dataset class to be used instead of default.
+        auto_generate_mask(bool): To automatically generate tile/WSI tissue mask
+            if is not provided.
+
+    """
+
+    def __init__(
+        self,
+        batch_size: int = 8,
+        num_loader_worker: int = 0,
+        num_postproc_worker: int = 0,
```

Comment:
Consider changing to 'works' e.g. 'num_works' as in the pytorch dataloader class.

Reply:
But pytorch use worker.

![image](https://user-images.githubusercontent.com/24943262/133756726-ba92a855-70d5-48ce-8252-7f2cc9d0af97.png)

Reply:
Do you mean the `s` ? will add them then

Reply:
Yeah, I jsut mean add an s to be consistent with where the phrase comes up elsewhere